ecl/hql/hqlexpr.cpp

```diff
@@ -8471,6 +8632,36 @@ inline bool canMergeDefinition(IHqlExpression * expr)
     return true;
 }
 
+IHqlExpression * CHqlMergedScope::lookupNearestSymbol(IIdAtom * searchId, unsigned lookupFlags, HqlLookupContext & ctx, NearestSymbol & nearest)
+{
+    CriticalBlock block(cs);
+    NearestSymbol tempNearest(searchId);
+    unsigned penalty = 0;
```

Comment:
This logic isn't correct.  I think it should be looping through mergedScopes, calling lookupNearest on each.